tools/run_tests/python_utils/jobset.py

```diff
@@ -490,4 +491,9 @@ def run(cmdlines,
     if remaining is not None:
       js.set_remaining(remaining)
   js.finish()
+
+  # Keep records of commands we have invoked for replaying later.
```

Comment:
I am opposed to introducing such special functionality deep into testing internals. The problem is we'd like to minimize touching the python_utils because it's risky (all tests depend on it) and it also seems unnecessary for obtaining the kind of information you'd need.